packages/@aws-cdk/integ-tests/lib/assertions/providers/provider.ts

```diff
@@ -115,13 +115,16 @@ class SingletonFunction extends CoreConstruct {
     super(scope, id);
     this.lambdaFunction = this.ensureFunction(props);
     this.serviceToken = this.lambdaFunction.serviceToken;
-  }
 
-  /**
-   * The policies can be added by different constructs
-   */
-  onPrepare(): void {
-    this.lambdaFunction.addPolicies(this.policies);
+    /**
+     * The policies can be added by different constructs
+     */
+    this.node.addValidation({
+      validate: () => {
+        this.lambdaFunction.addPolicies(this.policies);
+        return [];
+      },
+    });
```

Comment:
This needs to be an `Aspect`, not a validation.

Reply:
(for my own learning) What's the difference?

Reply:
It's when they run and what they're expected to do.

Aspects are allowed to change the construct tree, Validations are not.

Reply:
I see, that makes sense - thanks!